OpenRA.Mods.Common/Traits/RevealsMap.cs

```diff
@@ -44,7 +44,13 @@ protected void AddCellsToPlayerShroud(Actor self, Player p, PPos[] uv)
 			p.Shroud.AddSource(this, type, uv);
 		}
 
-		protected void RemoveCellsFromPlayerShroud(Actor self, Player p) { p.Shroud.RemoveSource(this); }
+		protected void RemoveCellsFromPlayerShroud(Actor self, Player p)
```

Comment:
IMO methods should do exactly what they say on the tin and callers should only call methods with the data they want modified, so this'd only be invoked for the players that should actually have cells removed from their shroud.

Reply:
I thought about that too, but wasn't sure because check was in AddCellsToPlayerShroud for the adding part. I'll go change it for both.